packages/react-component-library/src/components/DatePicker/DatePicker.tsx

```diff
@@ -185,6 +185,14 @@ export const DatePicker: React.FC<DatePickerProps> = ({
     }
   }
 
+  useEffect(() => {
```

Comment:
When I have been touching state I have been thinking about if it's a good time to abstract to a hook which feels like a pattern that has been working well. In this situation I wonder if there's an opportunity for a hook `useSelection` which manages `state` and how that would affect this code. It appears to be all manageable at this stage but I wonder if we add something else and the code starts becoming harder to manage.